pkg/apis/kops/validation/validation.go

```diff
@@ -289,6 +289,9 @@ func validateClusterSpec(spec *kops.ClusterSpec, c *kops.Cluster, fieldPath *fie
 		if !fi.ValueOf(spec.IAM.UseServiceAccountExternalPermissions) {
 			allErrs = append(allErrs, field.Forbidden(fldPath, "Karpenter requires that service accounts use external permissions"))
 		}
+		if spec.Karpenter.LogEncoding != "" {
+			allErrs = append(allErrs, field.Forbidden(fldPath.Child("logEncoding"), "logEncoding is deprecated"))
```

Comment:
Could be a warning, but as this was feature flagged LGTM

Reply:
Dropping this for now, as it may be re-added in https://github.com/kubernetes-sigs/karpenter/pull/2446.